api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -160,7 +160,7 @@ public DbUser updateUserWithRetries(
     int objectLockingFailureCount = 0;
     int statementClosedCount = 0;
     while (true) {
-      dbUser = userModifier.apply(dbUser);
+      dbUser = userDao.save(userModifier.apply(dbUser));
```

Comment:
If you call save here. I would move the try/catch up here and remove the save call below.

Reply:
thanks for taking a look - but local testing reveals that I am causing more problems because I don't know what I'm doing here.  So I'm going to drop this idea.